File: app/Http/Controllers/TasksController.php
<?php

namespace App\Http\Controllers;

use App\Http\Requests;
use App\Http\Controllers\Controller;
use App\Http\Controllers\FoldersController;
use App\Models\Folder;
use App\Models\Task;
use DB;
use Auth;

class TasksController extends Controller
{

	// checks current user auth status
	// if not auth, redirect to login
	public function __construct()
	{
		$this->middleware('auth');
	}

	public function index()
	{
		$user_id = Auth::user()->id;
		$data['page'] = 'tasks';
		$data['folders'] = FoldersController::getAll($data['page']);
		$data['selectedFolders'] = Folder::where(['user_id' => $user_id, 'page' => $data['page'], 'selected' => 1, 'active' => 0])->get();
		$data['activeFolder'] = Folder::where(['user_id' => $user_id, 'page' => $data['page'], 'active' => 1])->first();
		if ( $data['activeFolder'] != null ) {
			$data['activeFolderTasks'] = Task::where('folder_id', $data['activeFolder']->id)
				->orderBy('status')
				->orderBy('title', 'asc')->get();
		}
		foreach ($data['selectedFolders'] as $folder) {
			$data['selectedFolderTasks'][$folder->id] = Task::where('folder_id', $folder->id)
				->orderBy('status')
				->orderBy('title', 'asc')
				->get();
		}
		return view('tasks/content', $data);
	}

	public function activate($id)
	{
		Folder::where('user_id', 1)->where('active', 1)->update(['active' => 0]);
		Folder::where('user_id', 1)->where('id', $id)->update(['active' => 1]);
		return redirect('tasks');
	}

	public function status($id)
	{
		$task = Task::findOrFail($id);
		if ($task->status == 1) {
			$task->status = 0;
		} else {
			$task->status = 1;
		}
		$task->save();
		return redirect('tasks');
	}

	public function store(Requests\CreateTaskRequest $request)
	{
		$input = $request->all();
		$input['user_id'] = Auth::user()->id;
		Task::create($input);
		return redirect('tasks');
	}

	public function edit($id)
	{
		$data['page'] = 'tasks';
		$data['edit'] = true;
		$data['activeFolderId'] = $id;
		$data['action'] = '/tasks/update/' . $id;
		$data['folders'] = FoldersController::getAll($data['page']);
		$data['task'] = Task::findOrFail($id);
		return view('tasks/content', $data);
	}

	public function update($id, Requests\CreateTaskRequest $request)
	{
		$input = $request->all();
		$input['user_id'] = Auth::user()->id;
		$task = Task::where(['id' => $id, 'user_id' => $input['user_id']])->first();
		$task->update($input);
		return redirect('tasks');
	}

	public function clear($folder_id) {
		$user_id = Auth::user()->id;
		Task::where(['user_id' => $user_id, 'folder_id' => $folder_id, 'status' => 1])->delete();
		return redirect('tasks');
	}

}

File: app/php/TasksController.py


def index():
    user_id = request.user.id
    page = 'tasks'
    folders = FoldersController::getAll(page)
    selected_folders = Folder.objects.filter(user_id=user_id, page=page, selected=1, active=0).get()
    active_folder = Folder.objects.filter(user_id=user_id, page=page, active=1).first()

    if active_folder:
        active_folder_tasks = Task.objects.filter(folder_id=active_folder.id)
                .order_by('status')
                .order_by('title')
        for folder in selected_folders:
                selected_folder_tasks[folder.id] = Task.objects.filter('folder_id', folder.id)
                        .order_by('status')
                        .order_by('title')
                        .get()
        
    return view('tasks/content', context)


def activate(id):
    Folder.objects.filter(user_id=1, active=1).update(active=0)
    Folder.objects.filter(user_id=1, id=id).update(active=1)
    return redirect('/tasks/')


def status(id):
    task = Task::findOrFail(id)
    if task.status == 1:
        task.status = 0
    else: 
        task.status = 1
    task.save()
    return redirect('/tasks/')


def insert(Requests\CreateTaskRequest request):
    input = request.all()
    input['user_id'] = request.user.id
    Task::create(input)
    return redirect('/tasks/')


def edit(id):
    page = 'tasks'
    edit = true
    activeFolderId = id
    action = '/tasks/update/' . id
    folders = FoldersController::getAll(page)
    task = Task::findOrFail(id)
    return view('tasks/content', context)


def update(id, Requests\CreateTaskRequest request):
    input = request.all()
    input['user_id'] = request.user.id
    task = Task.objects.filter(id=id, user_id=input['user_id']).first()
    task.update(input)
    return redirect('/tasks/')


def clear(folder_id) :
    user_id = request.user.id
    Task.objects.filter(user_id=user_id, folder_id=folder_id, status=1).delete()
    return redirect('/tasks/')
